lib/active_model/serializer/adapter/json_api/error.rb

```diff
@@ -0,0 +1,75 @@
+require 'active_model_serializers/json_pointer'
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi < Base
+        module Error
+          UnknownSourceTypeError = Class.new(ArgumentError)
+
+          # @param [ActiveModel::Serializer::ErrorSerializer]
+          # @return [Array<Symbol, Array<String>] i.e. attribute_name, [attribute_errors]
+          def self.resource_errors(error_serializer)
+            error_serializer.as_json.flat_map do |attribute_name, attribute_errors|
+              attribute_error_objects(attribute_name, attribute_errors)
+            end
+          end
+
+          # definition:
+          #   JSON Object
+          #
+          # properties:
+          #   ☐ id      : String
+          #   ☐ status  : String
+          #   ☐ code    : String
+          #   ☐ title   : String
+          #   ☑ detail  : String
+          #   ☐ links
+          #   ☐ meta
+          #   ☑ error_source
+          #
+          # description:
+          #   id     : A unique identifier for this particular occurrence of the problem.
+          #   status : The HTTP status code applicable to this problem, expressed as a string value
+          #   code   : An application-specific error code, expressed as a string value.
+          #   title  : A short, human-readable summary of the problem. It **SHOULD NOT** change from
+          #     occurrence to occurrence of the problem, except for purposes of localization.
+          #   detail : A human-readable explanation specific to this occurrence of the problem.
+          def self.attribute_error_objects(attribute_name, attribute_errors)
+            attribute_errors.map do |attribute_error|
+              {
+                source: error_source(:pointer, attribute_name),
```

Comment:
Will the other properties (status, title, ...) be implemented as well? As an API consumer, having a title would help me group similar errors such as "Record Invalid" without having to determine what type of error is described in the detail "cannot be nil", "must be longer".